### PR TITLE
[New Model] Digital Product Pass

### LINKED ISSUE #183: [New Model]: Digital Product Passport
## Model Description
The "Digital Product Passport" is a data record that summarizes the components, materials and chemical substances or information on repairability, spare parts or professional disposal for a product. The data comes from all phases of the product life cycle and can be used for various purposes in all these phases (design, production, use, disposal).
Structuring environmental data in a standardized, comparable format enables all actors in the value and supply chain to work together towards a circular economy.
This data model is based on the new proposed Ecodesign Regulation (ESPR-2022) and is continuously adapted to the basic conditions over time.

## Relevant Standards
 link to ESPR regulation: https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX:52022AE0598

## Example Data
example of attributes:
- operatorIdentifier
- tradeName
- address
- contact
- localIdentifier
- facilityIdentifier
- status
- updateDate
- updateActivity
- description
- type
- physicalDimensions
- expectedLifetime
- expectedLifetimeUnit
- warranty
- recyclateContentName
- recyclateContentAmount
- sparePartName
- sparePartLocalIdentifier
- sparePartProducer
- toolAndProcessDocument
- assemblyAndDisassemblyDocument
- repairAndInstallationDocument
- userManualDocument
- EndOfLifeDocument
- returnAndDisposalDocument
- substanceOfConcernName
- substanceOfConcernLocation
- substanceOfConcernConcentration
- substanceOfConcernDocument
- disassemblySubstanceOfConcernDocument
- safeTransportationSubstanceOfConcernDocument
- safePackagingSubstanceOfConcernDocument
- otherSubstanceOfConcernDocument
- carbonFootprint
- carbonFootprintDocument
- carbonFootprintStandard
- carbonFootprintRule
- declarationOfConformityDocument
- complianceDocument
- criticalRawMaterials
- dateOfManufacture
- economicOperatorsRegistrationAndIdentificationNumber
- otherIdentification
- GlobalTradeIdentificationNumber
- commodityCode
- expectedWasteGenerationDocument
- packagingReuseSystem
- productPackahingDocument
- positioningDocument
- validityPeriod
- dataCarrier
- version
- additionalInformation
- renewableContentName
- renewableContentPercentage
- wastePreventionDocument
- sustainabilityInformationDocument

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example

### PR DESCRIPTION
**Pull request by catroest**
## Description

 -->

Closes #183 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the BAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar bamm-cli.jar -i \<path-to-aspect-model\> -v ). The  BAMM CLI is available [here](https://openmanufacturingplatform.github.io/sds-documentation/sds-developer-guide/dev-snapshot/tooling-guide/bamm-cli.html) and in [GitHub](https://github.com/OpenManufacturingPlatform/sds-sdk/releases)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "name" and "description"** in English language. 
- [x] **no duplicate names or preferredNames** within an Aspect (e.g. a Property and the referenced Characteristic should not have the same name)
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the BAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://openmanufacturingplatform.github.io/sds-documentation/bamm-specification/v1.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'

### CODE REVIEW

**Comment by matbmoser** [comment]
Looks awesome 👍🏻

File: io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl [hunk 1]
```diff
@@ -0,0 +1,846 @@
...
+:ownerId a bamm:Property;
+    bamm:preferredName "Owner Id"@en;
+    bamm:description "The identification of the owner of the product and the passport."@en;
+    bamm:characteristic :CompanyIdentification;
+    bamm:exampleValue "BPNL1234567890ZZ".
+:eori a bamm:Property;
+    bamm:preferredName "EORI"@en;
+    bamm:description "Any economic operator established in the customs territory of the Union needs, for customs purposes, an EORI number. EORI stands for economic operators registration and identification number."@en;
+    bamm:characteristic :EoriTrait;
+    bamm:exampleValue "DE 123 456 789 12345".
```
Comment (bs-jokri) [inline io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl:565]: have you checked that the example validates against the defined regex constraint?

File: io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl [hunk 2]
```diff
@@ -0,0 +1,846 @@
...
+:importer a bamm:Property;
+    bamm:preferredName "Importer"@en;
+    bamm:description "The importer of the product, if different from the owner of the passport."@en;
+    bamm:characteristic :CompanyIdentification;
+    bamm:exampleValue "BPNL1234567890ZZ".
+:gtin a bamm:Property;
+    bamm:preferredName "GTIN"@en;
+    bamm:description "GTIN stands for Global Trade Item Number. It is a unique and internationally recognized identifier used to identify products in the global marketplace. GTINs are typically encoded in barcodes and are used by retailers, manufacturers, and suppliers to track and manage their products throughout the supply chain. Valid inputs are all GTIN versions (GTIN-8, GTIN-12, GTIN-13, and GTIN-14)"@en;
+    bamm:characteristic :GtinTrait;
+    bamm:exampleValue "GTIN-8: 12345678".
```
Comment (bs-jokri) [inline io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl:315]: habe you checked if it validates against the regex?

**Review by bs-jokri — CHANGES_REQUESTED** [review]
please check the two regex examples and fix the whitespaces

File: io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl [hunk 3]
```diff
@@ -0,0 +1,846 @@
...
+    bamm:description "Characteristic for the expected product lifetime (lifespan)."@en;
+    bamm:dataType :LifespanEntity.
+:placedOnMarket a bamm:Property;
+    bamm:preferredName "Placed on Market"@en;
+    bamm:description "Date on  when the product is put in the market."@en;
+    bamm:characteristic :TimeStamp;
+    bamm:exampleValue "2000-01-01"^^xsd:date.
+:warranty a bamm:Property;
+    bamm:preferredName "Warranty"@en;
+    bamm:description "Commercial warranty period of the product (coupled to the Vehicle) in months. Durability and reliability of the product or its components as expressed through the product’s guaranteed lifetime, technical lifetime, mean time between failures, indication of real use information on the product, resistance to stresses or ageing mechanisms."@en;
```
[the remark this answers is not in the document]
Comment (catroest) [reply io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl:282]: fixed

File: io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl [hunk 4]
```diff
@@ -0,0 +1,846 @@
...
+    bamm-c:baseCharacteristic bamm-c:Text;
+    bamm-c:constraint :TitleConstraint.
+:ActivityEnumeration a bamm-c:Enumeration;
+    bamm:preferredName "Activity Enumeration"@en;
+    bamm:description "The activities of putting the product into service in form of an enumeration with the values Inspection, Maintenance, Repair, Preventive Maintenance, Accident and Other."@en;
+    bamm:dataType xsd:string;
+    bamm-c:values ("Inspection" "Maintenance" "Repair" "Preventive Maintenance" "Accident" "Other").
+:Either a bamm-c:Either;
+    bamm:preferredName "Either"@en;
+    bamm:description "Characteristic with two possible  values."@en;
```
[the remark this answers is not in the document]
Comment (catroest) [reply io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl:515]: fixed

File: io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl [hunk 5]
```diff
@@ -0,0 +1,846 @@
...
+    bamm:description "Constraint to limit the input to a valid ECLASS number. The constraint ensures that each pair of digits is followed by a hyphen, resulting in a pattern like \"00-00-00-00\" where the zeros can be replaced by any digit."@en;
+    bamm:value "^\\d{2}-\\d{2}-\\d{2}-\\d{2}$".
+:TitleConstraint a bamm-c:LengthConstraint;
+    bamm:preferredName "Title Constraint"@en;
+    bamm:description "A constraint that limits the title to 100 characters."@en;
+    bamm-c:minValue "3"^^xsd:nonNegativeInteger;
+    bamm-c:maxValue "100"^^xsd:nonNegativeInteger.
+:ConcentrationEnumeration a bamm-c:Enumeration;
+    bamm:preferredName "Concentration Enumeration"@en;
+    bamm:description "Enumeration of possible units of concentration with percent, volume percent, parts per thousand, parts per million, parts per billion and  parts per trillion."@en;
```
[the remark this answers is not in the document]
Comment (catroest) [reply io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl:616]: fixed

File: io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl (hunk 1 above)
Comment (catroest) [reply io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl:565]: fixed example and regex

File: io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl (hunk 2 above)
Comment (catroest) [reply io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl:315]: fixed example (deleted start)

File: io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl [hunk 6]
```diff
@@ -0,0 +1,846 @@
...
+    bamm:preferredName "Content"@en;
+    bamm:description "The content of the document e.g a link."@en;
+    bamm:characteristic bamm-c:Text;
+    bamm:exampleValue "www.alink.pdf".
+:category a bamm:Property;
+    bamm:preferredName "Category"@en;
+    bamm:description "The category in which the document can be sorted."@en;
+    bamm:characteristic :CategoryEnumeration;
+    bamm:exampleValue "Sustainability".
+:contentType a bamm:Property;
```
[the remark this answers is not in the document]
Comment (catroest) [reply io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl:215]: There are sadly too many document types to make an enumeration and different understandings what to expect. This is just an open field. But I will discuss it with the team.

**Comment by catroest** [comment]
the model validates with 2.1.3 cli

**Comment by catroest** [comment]
Modeling_Team

File: io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl (hunk 6 above)
[the remark this answers is not in the document]
Comment (catroest) [reply io.catenax.digital_product_passport/1.0.0/DigitalProductPassport.ttl:275]: will stay as open field

**Review by bs-jokri — APPROVED** [review]
LGTM

**Pull request merged**